### PR TITLE
Add --db-port option

### PR DESCRIPTION
**Pull request by geichelberger**
Add support for non-default database ports. For example, you cannot run a MariaDB and MySQL database on the same host with the default ports.

### CODE REVIEW

**Comment by kabalin** [comment]
Hi @geichelberger, thanks for your contribution. Please reflect changes in docs/CHANGELOG.md also.

**Comment by stronk7** [comment]
HI,

I just found myself coding exactly the same changes performed by @geichelberger here.

The reason is that, with Moodle 4.1, we have raised the PostgreSQL requirements to version 12. And, while that should be a trivial change (and it is for GHA), for Travis they decided to change the default user and the default port for PostgreSQL >= 11.

Link: https://docs.travis-ci.com/user/database-setup/#using-a-different-postgresql-version

> For PostgreSQL 11 and 12 you must specify the packages, but the user is travis and the port is 5433 instead. So you must specify the PGPORT

I already have a PR near finished that, pseudo-automatically (based on an env variable called `PGVER`) decides between:
- db-user: postgres or travis.
- db-port: 5432 or 5433.

And the only piece that is missing... is exactly this PR, aka, add support for db-port (within db-options). Hence, it has my (self-interested) +1.

If you cannot do it, @geichelberger , would you mind if we add a commit to your PR branch amending the CHANGELOG, as @kabalin asked some days ago?

Ciao :-)

**Comment by geichelberger** [comment]
Hi, I was busy the last few days and planned to do it today ;)

**Comment by stronk7** [comment]
For the records, np about the master failures above (requiring PostgreSQL 12 for 4.1dev). That's exactly what my PR will be fixing, in the best possible way, trivial for GHA, but specifically for travis, where we'll be using this new db-port option to woraround the mess they did.

Ciao :-)

**Comment by stronk7** [comment]
Sorry, forgot to say thanks, @geichelberger !

File: res/template/config.php.txt [hunk 1]
```diff
@@ -11,7 +11,9 @@ $CFG->dbname    = '{{DBNAME}}';
 $CFG->dbuser    = '{{DBUSER}}';
 $CFG->dbpass    = '{{DBPASS}}';
 $CFG->prefix    = 'mdl_';
-$CFG->dboptions = [];
+$CFG->dboptions = [
+    'dbport' => '{{DBPORT}}',
```
Comment (scara) [inline res/template/config.php.txt:15]: Hi @geichelberger and @stronk7,
should be `dbport` a string or an integer?

Moodle [uses](https://github.com/moodle/moodle/blob/d9632ca8b28304d5188a8310480f0cd6bc06d9d8/config-dist.php#L59-L61) a string, wondering if it is just for an "easy" fallback to the default port which in turns it's an actual empty string in the CLI command.

HTH,
Matteo

Comment (kabalin) [reply res/template/config.php.txt:15]: I would say using string is OK here, as it does not seem important. It is responsibility of db engine class representing Moodle database interface to handle port param for connection: for MySQL it is [casted to int](https://github.com/moodle/moodle/blob/master/lib/dml/mysqli_native_moodle_database.php#L555), while for Postgres, it is [treated as string](https://github.com/moodle/moodle/blob/master/lib/dml/pgsql_native_moodle_database.php#L157).

**Review by kabalin — APPROVED** [review]
The change looks good to me.

One thing, rather than conditionally inject `--port` to connection string (in `src/Installer/Database/...`), we can have it permanently defined in the string and set specific to engine default port in the class variable along with type (which is then overridden if custom value is provided). Thinking more, in fact probably not the good idea in case if any of those clients can read port param from env variable, as having port always set via suggested change may break some installations...

Anyway, good to merge (if @scara is happy with my comment re type). Thanks @geichelberger!

**Comment by stronk7** [comment]
Some 👍 here would be really appreciated, @scara (only if it's ok for you, of course!). This is stopping one more incoming PR (4.1 requirements) and a release to be done immediately after that.

TIA!

**Comment by scara** [comment]
Hi @stronk7,
oops... My apologies, missed the question from @kabalin.

Here is my 👍 based on the comments above which makes me confident about the db-port-as-a-string usage.

HTH,
Matteo

**Comment by stronk7** [comment]
Thanks @scara and @geichelberger !

I'm going to merge this now on behalf of @kabalin (thanks too!).

Ciao :-)

**Comment by stronk7** [comment]
Merging this no matter Travis is failing with master branch. It's known issue (new Moodle 4.1 requirements) and will be fixed soon by another PR, based on this.

**Pull request merged**